bookmarklet.py

```diff
@@ -50,48 +45,32 @@ def add():
     url = request.args.get('url', None)
     if url is None:
         return 'You didn\'t pass in a url param!'
-    event_id = get_event_id(url)
-    if event_id is None:
-        return 'This isn\'t an eventbrite page, so we can\'t add it.'
-    session['event_id'] = event_id
+
+    url_details = urlparse(url)
+    base_url = url_details.netloc
+    if base_url in URL_HANDLERS:
+        handler = URL_HANDLERS[base_url]
+        data = handler(url)
+        session['data'] = data
+    else:
+        return 'This isn\'t a supported page, so we can\'t add it.'
 
     return render_template('add.html',
                             categories=build_contentful_data.get_categories(),
                             tags=build_contentful_data.get_tags())
 
 @app.route("/tag", methods=['GET'])
 def added():
-    event_id = session['event_id']
-    api_url = EVENTBRITE_API_BASE + event_id + '?expand=venue'
-    response = requests.get(
-        api_url,
-        headers = {
-            "Authorization": "Bearer %s" % EVENTBRITE_OAUTH_TOKEN,
-        },
-        verify = True,
-    )
-    title = response.json()['name']['text']
-    description = response.json()['description']['text']
-    start_time = response.json()['start']['local']
-    end_time = response.json()['end']['local']
-
-    external_url = EVENTBRITE_URL_BASE + event_id
-
-    # location data
-    lat = float(response.json()['venue']['address']['latitude'])
-    lon = float(response.json()['venue']['address']['longitude'])
-    name = response.json()['venue']['name']
-    room = response.json()['venue']['address']['localized_address_display']
-    location_id = build_contentful_data.find_location_id(lat, lon, name, room)
-
     tags = request.args.getlist('tag')
-    event_attributes = build_contentful_data.build_event(title,
-                                    start_time=start_time,
-                                    end_time=end_time,
-                                    description=description,
-                                    external_url=external_url,
-                                    location_id=location_id,
-                                    category=request.args.get('category', None),
+    category=request.args.get('category', None)
+    data = session['data']
```

Comment:
How does the code handle missing data?
Should you check for the presence of the data you need?